spec/controllers/videos_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'spec_helper'
+
+describe VideosController do 
+  context 'required authenticated user' do
+    before do
+      session[:user_id] = Fabricate(:user).id
+    end 
+    describe "GET show" do 
+      it "find the video we want to display the detail" do 
+        #setup data for test
+        video = Fabricate(:video)
+        get :show, id: video.id
+        expect(assigns(:video)).to eq(video) 
+        # the :video inside "assigns" is the instance variable @video created by the show action
+      end
+      it "sets @reviews for the video" do 
+        video = Fabricate(:video)
+        review1 = Fabricate(:review, video_id: video.id) #if I use video: video here, it will fail
+        review2 = Fabricate(:review, video_id: video.id)
+        get :show, id: video.id
+        expect(assigns(:reviews)).to match_array([review1, review2])
+      end
+      #Here we do not need to test the render function, because it's part of rails convention
```

Comment:
You should make these kind of notes to a blog post, and code should be just code not a blog.